src/WebGPU/Silk.NET.WebGPU/Platforms/MacOS/Selector.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Silk.NET.Core.Native;
+
+namespace Silk.NET.WebGPU.Platforms.MacOS;
+
+internal struct Selector
+{
+    public readonly nint NativePtr;
+
+    public Selector(nint ptr)
+    {
+        NativePtr = ptr;
+    }
+
+    public Selector(string name)
+    {
+        NativePtr = ObjectiveCRuntime.sel_registerName(SilkMarshal.StringToPtr(name));
```

Comment:
this string should be freed somewhere, as this is a memory leak

Reply:
Ah, my bad. I will address that.

Reply:
Fixed both instances.